pymc/gp/hsgp_approx.py

```diff
@@ -171,20 +199,39 @@ def __init__(
             raise ValueError(arg_err_msg)
         m = tuple(m)
 
-        if (L is None and c is None) or (L is not None and c is not None):
-            raise ValueError("Provide one of `c` or `L`")
-
-        if L is not None and (not isinstance(L, Sequence) or len(L) != cov_func.n_dims):
-            raise ValueError(arg_err_msg)
-
-        if L is None and c is not None and c < 1.2:
-            warnings.warn("For an adequate approximation `c >= 1.2` is recommended.")
+        if isinstance(cov_func, Periodic):
+            if cov_func.n_dims > 1:
+                raise ValueError(
+                    "HSGP approximation for `Periodic` kernel only implemented for 1-dimensional case."
+                )
+            if (L or c or parameterization) is not None:
+                warnings.warn(
+                    "Argument `L`, `c` or `parameterization` supplied but not used for `Periodic` kernel."
```

Comment:
I think this warning will show up a lot, would add something like "For this kernel `m` is the only parameter controlling the approximation that must be supplied."

Reply:
Also am seeing this warning always go off, even if I just pass `m`, because the default value of `parameterization` is `noncentered`.   Should change the triggering clause to `(L or c is not None) or (parameterization == "centered")`